ruby/bob.rb

```diff
@@ -0,0 +1,37 @@
+class Bob
+  def hey(remark)
+    @remark = Remark.new(remark)
+    return "Fine. Be that way!" if @remark.is_blank?
+    return "Whoa, chill out!" if @remark.is_shouting?
+    return "Sure." if @remark.is_question?
+    "Whatever."
+  end
+
+end
+
+class Remark
+  attr_reader = :remark
+
+  def initialize(remark)
+    @remark = remark
+  end
+
+  def is_question?
+    @remark[-1] == "?"
+  end
+
+  def is_shouting?
+    return false if self.is_only_numbers?
+    @remark == @remark.upcase
+  end
+
+  def is_only_numbers?
+    no_words = /[^a-zA-Z]*[?]*/
```

Comment:
It is more accurate that I am checking for "has_no_words", so I changed the name of this method. After looking into it, I am not seeing a way to check for "only numbers, punctuation, or whitespace" that is more friendly than this.